src/api/app/models/event/request.rb

```diff
@@ -134,8 +134,8 @@ def source_watchers
 
     def find_watchers(project_key)
       project_names = payload['actions'].map { |action| action[project_key] }.uniq
-      projects = Project.where(name: project_names).joins(watched_projects: :user)
-      projects.flat_map { |project| project.watched_projects.map(&:user) }
+      projects = WatchedProject.where(project: Project.where(name: project_names))
```

Comment:
Small point but please change the variable name from `projects` to `watched_projects`.